src/algebraic_geometry/elliptic_curve/point.lean

```diff
@@ -94,54 +94,35 @@ variables {F : Type u} [comm_ring F] (W : weierstrass_curve F) (x x₁ x₂ y y
 noncomputable def neg_polynomial : _root_.polynomial $ _root_.polynomial F :=
 -X - C (C W.a₁ * X + C W.a₃)
 
-lemma Y_add_neg_polynomial : X + W.neg_polynomial = -C (C W.a₁ * X + C W.a₃) :=
-by { rw [neg_polynomial], ring1 }
-
-lemma Y_sub_neg_polynomial : X - W.neg_polynomial = W.polynomial_Y :=
-by { rw [neg_polynomial, polynomial_Y], C_simp, ring1 }
-
-lemma Y_mul_neg_polynomial :
-  X * W.neg_polynomial = -C (X ^ 3 + C W.a₂ * X ^ 2 + C W.a₄ * X + C W.a₆) - W.polynomial :=
-by { rw [neg_polynomial, polynomial], ring1 }
-
-lemma coordinate_ring.Y_mul_neg_polynomial :
-  adjoin_root.mk W.polynomial (X * W.neg_polynomial)
-    = adjoin_root.mk W.polynomial (-C (X ^ 3 + C W.a₂ * X ^ 2 + C W.a₄ * X + C W.a₆)) :=
-adjoin_root.mk_eq_mk.mpr ⟨-1, by rw [Y_mul_neg_polynomial, sub_sub_cancel_left, mul_neg_one]⟩
-
 /-- The $Y$-coordinate of the negation of an affine point. -/
 @[simp] def neg_Y : F := -y - W.a₁ * x - W.a₃
 
 lemma neg_Y_neg_Y : -W.neg_Y x y - W.a₁ * x - W.a₃ = y := by { rw [neg_Y], ring1 }
 
-lemma neg_Y_eq_eval : W.neg_Y x y = eval x (eval (C y) W.neg_polynomial) :=
+@[simp] lemma eval_neg_polynomial : eval x (eval (C y) W.neg_polynomial) = W.neg_Y x y :=
 by { rw [neg_Y, sub_sub, neg_polynomial], eval_simp }
 
-/-- The polynomial obtained by substituting the line $Y := L*(X - x_1) + y_1$, with a slope of $L$
-and contains a point $(x_1, y_1)$ of `W`, into the polynomial $W(X, Y)$ associated to `W`.
+/-- The polynomial $L*(X - x) + y$ associated to the line $Y = L*(X - x) + y$,
+with a slope of $L$ that passes through an affine point $(x, y)$.
+This does not depend on `W`, and has the argument order $x$ and $y$. -/
+noncomputable def line_polynomial : _root_.polynomial F := C L * (X - C x) + C y
```

Comment:
You won't need the `_root_` if you make [weierstrass_curve.polynomial](https://leanprover-community.github.io/mathlib_docs/algebraic_geometry/elliptic_curve/weierstrass.html#weierstrass_curve.polynomial) protected, right?

Reply:
Yeah, great idea, thanks

Reply:
Hmm replacing `polynomial` by `[X]` is obviously a better solution (and what you did), but protecting seems a good idea anyway.

Reply:
Without protecting it, I can't do `F[X]` because of some weird bug with locales and it wants to take in a `weierstrass_curve`.

Reply:
Oh strange, thanks for letting me know!